align/cell_fabric/drc.py

```diff
@@ -35,6 +36,38 @@ def _check_via_rules(self, layer, vv):
         space = self.canvas.pdk[layer]['SpaceX']
         return space
 
+    def _check_via_enclosure_rules(self, layer, vv):
+        '''Check via enclosures.'''
+        v = self.canvas.pdk[layer]
+        [ly_l,ly_u] = v['Stack']
+        if ly_l is not None:
+            ml_dir = self.canvas.pdk[ly_l]['Direction'].upper()
+            assert ml_dir in ['V','H']
+        if ly_u is not None:
+            mu_dir = self.canvas.pdk[ly_u]['Direction'].upper()
+            assert mu_dir in ['V','H']
+
+        def check_single_metal( r, ly, metal_dir, enclosure_value):
+            o = 1 if metal_dir == 'V' else 0
+            metal_r = self._find_rect_covering_via( r, ly, metal_dir)
+            if metal_r[o+0] > r.rect[o+0] - enclosure_value or metal_r[o+2] < r.rect[o+2] + enclosure_value:
+                self.errors.append( f"ENCLOSURE: {ly} {metal_r} does not sufficiently surround {layer} {r.rect}, {enclosure_value}")
+
+        for _, sl in vv.items():
+            for r in sl.rects:
+                if ly_l is not None: check_single_metal( r, ly_l, ml_dir, v['VencA_L'])
+                if ly_u is not None: check_single_metal( r, ly_u, mu_dir, v['VencA_H'])
```

Comment:
The issue may lie here.

check_single_metal looks like its checking the the enclosure perpendicular to the direction of the metal. But VencA_* refers to the enclosure along the direction of the metal. We probably want to check both rules correct?

Reply:
Ignore this comment. You are checking VencA_*. There is a real issue in the enclosure. Looking into it.

Reply:
Think it should be fixed now.